### PR TITLE
Decoding of strings with PIDs fails

### PR DESCRIPTION
**Pull request by kubaodias**

### CODE REVIEW

File: src/code_transform.erl [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+%%------------------------------------------------------------------------------
+%% @spec traverse(S, F) -> Text
+%% where
+%%		S = any()
+%%		F = Fun
+%%		Text = list()
+%% @doc Traverses structure, executes F and returns not flattened list
+%% @end
+%%------------------------------------------------------------------------------
+traverse(L, F) when is_list(L) ->
```
Comment (sjanota) [inline src/code_transform.erl]: Specs maybe?

File: src/code_transform.erl [hunk 2]
```diff
@@ -0,0 +1,139 @@
+%%------------------------------------------------------------------------------
+%% @author kuba.odias
+%% @copyright relayr 2009-2018
+%% @doc Functions for traversing of Erlang terms.
+%% @end
+%%------------------------------------------------------------------------------
+-module(code_transform).
```
Comment (sjanota) [inline src/code_transform.erl]: According to description module is meant to contain `Functions for traversing of Erlang terms.`. Why not `code_traverse` then?

File: src/code_transform.erl [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+%% Include files
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% Function exports
+%%------------------------------------------------------------------------------
+-export([
+    traverse/1,
+    traverse/2
+]).
```
Comment (sjanota) [inline src/code_transform.erl]: If those functions are meant to be used outside of this library I would also reexport them in `code_utils` module. Thanks to that client have single entry point for whole library.

File: src/code_transform.erl [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+traverse(T) ->
+    traverse(T, fun simple_transform/2).
+
+%%------------------------------------------------------------------------------
+%% @spec traverse(S, F) -> Text
+%% where
+%%		S = any()
+%%		F = Fun
+%%		Text = list()
+%% @doc Traverses structure, executes F and returns not flattened list
```
Comment (sjanota) [inline src/code_transform.erl]: I don't really understand what this function does. Can this doc be more elaborate? Maybe an example?

File: src/code_transform.erl [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+%% Macros
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% Records
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% Types
+%%------------------------------------------------------------------------------
```
Comment (sjanota) [inline src/code_transform.erl]: Do those sections add any value to the file except making it longer? Maybe they can be cleaned up?

**Comment by kubaodias** [comment]
Refactoring done in d24d1c8. Please review again.

File: src/code_transform.erl [hunk 6]
```diff
@@ -1,102 +1,52 @@
...
+%% Types
 %%------------------------------------------------------------------------------
+-type transform_value_type():: atom | number | pid.
+-type transform_action()    :: list_start | list_stop | binary_start | binary_stop | tuple_start | tuple_stop | {value, transform_value_type()}.
+-type transform_output()    :: {go, integer() | list(), list()} | {stop, list()} | integer() | list().
 
-%%------------------------------------------------------------------------------
-%% Records
-%%------------------------------------------------------------------------------
+-type transform_function()  :: fun((Action :: transform_action(), Term :: term()) -> transform_output()).
```
Comment (sjanota) [inline src/code_transform.erl]: Ok, this part helps a lot with understanding what's going on. Thanks!

**Pull request merged**